includes/admin/class-wc-rest-payments-settings-controller.php

```diff
@@ -179,4 +198,43 @@ private function update_enabled_payment_methods( WP_REST_Request $request ) {
 		$this->wcpay_gateway->update_option( 'payment_method_order', $payment_method_ids_to_enable );
 	}
 
+	/**
+	 * Updates the digital wallets enable/disable settings.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 */
+	private function update_is_digital_wallets_enabled( WP_REST_Request $request ) {
+		if ( ! $request->has_param( 'is_digital_wallets_enabled' ) ) {
+			return;
+		}
+
+		$is_digital_wallets_enabled = $request->get_param( 'is_digital_wallets_enabled' );
+
+		if ( $is_digital_wallets_enabled ) {
+			$this->digital_wallets_gateway->enable();
+		} else {
+			$this->digital_wallets_gateway->disable();
+		}
+	}
+
+	/**
+	 * Updates the list of sections that will show digital wallets.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 */
+	private function update_digital_wallets_enabled_sections( WP_REST_Request $request ) {
+		if ( ! $request->has_param( 'digital_wallets_enabled_sections' ) ) {
+			return;
+		}
+
+		$digital_wallets_enabled_sections         = $request->get_param( 'digital_wallets_enabled_sections' );
+		$updated_digital_wallets_enabled_sections = [];
+
+		$remove_invalid_sections                  = function ( $section ) {
```

Comment:
It looks like we would be getting the validation functionality "for free" if we used the validation on the `register_rest_route` on L74 - what do you think about doing that instead? That's what `enabled_payment_method_ids` is doing as well, to validate the payment method names.

Reply:
Good callout. The validation should be done on the Controller. In this case, `register_rest_route` is a better place for it.

Reply:
Updated. The controller will now validate the `digital_wallets_enabled_sections` object.